pkg/controller/bundle/bundle_unpacker.go

```diff
@@ -165,6 +171,21 @@ func (c *ConfigMapUnpacker) job(cmRef *corev1.ObjectReference, bundlePath string
 	job.SetName(cmRef.Name)
 	job.SetOwnerReferences([]metav1.OwnerReference{ownerRef(cmRef)})
 
+	// Don't set a timeout if it is 0
+	if c.unpackTimeout != time.Duration(0) {
+		t := int64(c.unpackTimeout.Seconds())
+		job.Spec.ActiveDeadlineSeconds = &t
+	}
+
+	// By default the BackoffLimit is set to 6 which with exponential backoff 10s + 20s + 40s ...
+	// translates to ~10m of waiting time.
+	// We want to fail faster than that when we have repeated failures from the bundle unpack pod
+	// so we set it to 4 which is ~2.5m of waiting time
+	// See: https://kubernetes.io/docs/concepts/workloads/controllers/job/#pod-backoff-failure-policy
+	// TODO (haseeb): Should this be configurable as well?
+	backOffLimit := int32(4)
+	job.Spec.BackoffLimit = &backOffLimit
```

Comment:
@benluddy Does this seem reasonable if we want to fail fast in the case of repeated crashes, e.g the `pull` or `extract` containers failing due to invalid bundle contents.

Ideally we'd fail after the first time on a fatal error like that, but it's hard to tell if there wasn't some other transient error that might be remedied by a restart.

Reply:
That seems reasonable to me. The extract container needs to talk to the API server to write to a ConfigMap, so there's at least some risk of transient error. I wouldn't object to an even lower limit (three attempts feels good).

I don't think it's worth it, but we could always instrument unpacking to report metrics on bundle unpacking (time to completion, number of attempts, outcome) and try to gauge how useful retries are.

Reply:
I've bumped the number of retries down to 3.
@benluddy Do you think it might be worth making this configurable? Not sure it would be much help for fatal errors. And I guess 3 is good enough for transient errors as well.